### DISCUSSION TITLE
[csl] avoid retransmissions of a frame with CSL IE

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#7825] [Q&A] [answered]
As described in this issue [6976](https://github.com/openthread/openthread/issues/6976) , For retransmissions of a frame with CSL IE, the phase is set just on first try (and the frame secured).
For the next tries, the frame remains the same, having invalid phase.

so why can't we just prevent retransmissions of a frame with CSL IE so it would create new frame again with correct phase ?

**Comment by jwhui** [comment, marked as answer]
The Thread Specification was updated to address this issue - see [SPEC-977](https://threadgroup.atlassian.net/browse/SPEC-977) for details.

We still need to implement the latest specified behavior. I have included it as a TODO on the [Thread Spec Implementation project page](https://github.com/orgs/openthread/projects/1/).

**Comment by testter2** [comment]
Thanks @jwhui
